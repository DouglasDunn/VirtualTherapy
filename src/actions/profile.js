import database from '../firebase/firebase';

// CREATE_PROFILE
export const createProfile = (profile) => ({
  type: 'CREATE_PROFILE',
  profile
});

export const startCreateProfile = (profileData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      firstName = '',
      lastName = '',
      emailAddress = '',
      dateOfBirth = '',
      gender = '',
      emergencyContactName = '',
      emergencyContactNumber = ''
    } = profileData;
    const profile = { firstName, lastName, emailAddress, dateOfBirth, gender, emergencyContactName, emergencyContactNumber };

    return database.ref(`users/${uid}/profile`).set(profile).then((ref) => {
      dispatch(createProfile({
        ...profile
      }));
    });
  };
};

// SET_PROFILE
export const setProfile = (profile) => {
  if (profile === null) {
    profile = {};
  }

  return {
    type: 'SET_PROFILE',
    profile
  };
};

export const startSetProfile = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/profile`).once('value').then((snapshot) => {
      dispatch(setProfile(snapshot.val()));
    });
  };
};
