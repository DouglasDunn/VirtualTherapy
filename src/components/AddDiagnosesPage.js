import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
// import { startAddExpense } from '../actions/expenses';

export class AddDiagnosesPage extends React.Component {
  // onSubmit = (expense) => {
  //   this.props.startAddExpense(expense);
  //   this.props.history.push('/');
  // };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm />
        </div>
      </div>
    );
  }
}

// <ExpenseForm
//   onSubmit={this.onSubmit}
// />

// const mapDispatchToProps = (dispatch) => ({
//   startAddExpense: (expense) => dispatch(startAddExpense(expense))
// });

export default AddDiagnosesPage;
//export default connect(undefined, mapDispatchToProps)(AddExpensePage);
