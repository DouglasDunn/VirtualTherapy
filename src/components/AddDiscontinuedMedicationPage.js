import React from 'react';

export class AddDiscontinuedMedicationPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      discontinuedMedication: '' ? 'yes' : 'no'
    };
  }

  onDiscontinuedMedicationChange = (e) => {
    const discontinuedMedication = e.target.value;
    this.setState(() => ({ discontinuedMedication }));
  };

  handlePrevious = () => {
    this.props.history.push('/add-medication');
  };

  onSubmit = (e) => {
    e.preventDefault();

    this.props.history.push('/exercise-questions');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Discontinued Medications</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            Have you stopped taking any medications due to side effects?
            <label>
              <input
                type="radio"
                checked={this.state.discontinuedMedication === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onDiscontinuedMedicationChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.discontinuedMedication === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onDiscontinuedMedicationChange}
              />
              No
            </label>
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

export default AddDiscontinuedMedicationPage;
