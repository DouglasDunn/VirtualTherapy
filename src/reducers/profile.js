// Profiles Reducer

const profileReducerDefaultState = {};

export default (state = profileReducerDefaultState, action) => {
  switch (action.type) {
    case 'CREATE_PROFILE':
      return {
        ...action.profile
      };
    case 'EDIT_PROFILE':
      return {
        ...state,
        ...action.updates
      };
    case 'SET_PROFILE':
      return action.profile;
    default:
      return state;
  }
};
