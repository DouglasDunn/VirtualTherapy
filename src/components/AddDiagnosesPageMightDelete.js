import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
import { startAddDiagnoses } from '../actions/diagnoses';
import DiagnosesListItem from './DiagnosesListItem';

export class AddDiagnosesPage extends React.Component {
  onSubmit = (diagnoses) => {
    this.props.startAddDiagnoses(diagnoses);
    this.props.history.push('/diagnoses');
    console.log('add');
    console.log(this.props);
  };
  // nextPage = () => {
  //   this.props.history.push('/diagnoses');
  // };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

// const mapStateToProps = (state) => ({
//   diagnoses: state.diagnoses
// });

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

// export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);

export default connect(undefined, mapDispatchToProps)(AddDiagnosesPage);

// <div className="content-container">
//   <div className="list-header">
//     <div className="show-for-mobile">Diagnoses</div>
//     <div className="show-for-desktop">Diagnoses</div>
//     <div className="show-for-desktop">Time</div>
//   </div>
//   <div className="list-body">
//     {
//       this.props.diagnoses.length === 0 ? (
//         <div className="list-item list-item--message">
//           <span>No Diagnoses yet, click the button above to add them.</span>
//         </div>
//       ) : (
//         this.props.diagnoses.map((diagnoses) => {
//           return <DiagnosesListItem key={diagnoses.id} {...diagnoses} />;
//         })
//       )
//     }
//   </div>
// </div>
// <div className="content-container">
//   <button onClick={this.nextPage} className="button">Next</button>
// </div>
