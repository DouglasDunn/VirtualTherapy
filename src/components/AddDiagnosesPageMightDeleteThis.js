import React from 'react';
import { connect } from 'react-redux';
import { startAddDiagnoses } from '../actions/diagnoses';

const initialErrorState = {
  diagnosedError: '',
  diagnosesListError: ''
};

export class AddDiagnosesPage extends React.Component {
  constructor(props) {
    super(props);
    let radioButton = 'no';
    let diagnosesArray;
    if (this.props.diagnoses) {
      if (Array.isArray(this.props.diagnoses)) {
        radioButton = 'yes';
        console.log(Array.isArray(this.props.diagnoses));
        console.log(this.props.diagnoses);
        diagnosesArray = this.props.diagnoses.map(({ diagnoses }) => diagnoses).join(', ');
      }
    }

    this.state = {
      diagnosed: this.props.diagnoses ? radioButton: '',
      diagnosesList: this.props.diagnoses ? diagnosesArray : '',
      ...initialErrorState
    };
  }

  onDiagnosedChange = (e) => {
    const diagnosed = e.target.value;
    this.setState(() => ({ diagnosed }));
  };

  onDiagnosesListChange = (e) => {
    const diagnosesList = e.target.value;
    this.setState(() => ({ diagnosesList }));
  };

  validate = () => {
    let diagnosedError = '';
    let diagnosesListError = '';

    if (!this.state.diagnosed) {
      diagnosedError = 'Please choose an option';
    }

    if ((this.state.diagnosed === 'yes') && !this.state.diagnosesList) {
      diagnosesListError = 'diagnoses list cannot be blank';
    } else {
      diagnosesListError = '';
    }

    if (diagnosedError || diagnosesListError) {
      this.setState({ diagnosedError, diagnosesListError });
      return false;
    }

    return true;
  };

  handlePrevious = () => {
    this.props.history.push('/add-drug-allergies');
  };

  onSubmit = (e) => {
    e.preventDefault();
    const isValid = this.validate();
    if (isValid) {
      if (this.state.diagnosed === 'no') {
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: ''
        });
      } else {
        console.log(this.state.diagnosesList);
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: this.state.diagnosesList
        });
      }
      this.props.history.push('/add-medication');
    }
  };

  renderDiagnosesList = () => {
    if (this.state.diagnosed === 'yes') {
      return (
        <div>
          {this.state.diagnosesListError && <p className="form__error">{this.state.diagnosesListError}</p>}
          <textarea
            placeholder="Type in all your symptoms and diagnoses separated by a comma"
            className="textarea"
            value={this.state.diagnosesList}
            onChange={this.onDiagnosesListChange}
          >
          </textarea>
        </div>
      );
    }
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            {this.state.diagnosedError && <p className="form__error">{this.state.diagnosedError}</p>}
            Has a doctor ever diagnosed you with a mental health disease?
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onDiagnosedChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onDiagnosedChange}
              />
              No
            </label>
            {this.renderDiagnosesList()}
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  diagnoses: state.diagnoses
});

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);
