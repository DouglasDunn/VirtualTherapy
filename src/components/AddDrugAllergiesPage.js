import React from 'react';
import { connect } from 'react-redux';
import DrugAllergyForm from './DrugAllergyForm';
import { startAddDrugAllergy } from '../actions/drugAllergy';

export class AddDrugAllergiesPage extends React.Component {
  onSubmit = (drugAllergy) => {
    this.props.startAddDrugAllergy(drugAllergy);
    this.props.history.push('/add-diagnoses');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Drug Allergies</h1>
          </div>
        </div>
        <div className="content-container">
          <DrugAllergyForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddDrugAllergy: (drugAllergy) => dispatch(startAddDrugAllergy(drugAllergy))
});

export default connect(undefined, mapDispatchToProps)(AddDrugAllergiesPage);
