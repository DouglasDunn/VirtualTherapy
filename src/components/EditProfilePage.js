import React from 'react';
import { connect } from 'react-redux';
import ProfileForm from './ProfileForm';
import { startEditProfile } from '../actions/profile';

export class EditProfilePage extends React.Component {
  onSubmit = (profile) => {
    this.props.startEditProfile(profile);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Profile</h1>
          </div>
        </div>
        <div className="content-container">
          <ProfileForm
            profile={this.props.profile}
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  profile: state.profile
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditProfile: (profile) => dispatch(startEditProfile(profile))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditProfilePage);
