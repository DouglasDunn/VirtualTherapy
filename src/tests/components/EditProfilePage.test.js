import React from 'react';
import { shallow } from 'enzyme';
import profile from '../fixtures/profile';
import { EditProfilePage } from '../../components/EditProfilePage';

let startEditProfile, history, wrapper;

beforeEach(() => {
  startEditProfile = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditProfilePage
      startEditProfile={startEditProfile}
      history={history}
      profile={profile}
    />
  );
});

test('should render EditProfilePage', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle startEditProfile', () => {
  wrapper.find('ProfileForm').prop('onSubmit')(profile);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(startEditProfile).toHaveBeenLastCalledWith(profile);
});
