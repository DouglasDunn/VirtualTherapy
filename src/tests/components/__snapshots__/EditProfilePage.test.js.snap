// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditProfilePage 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Profile
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ProfileForm
      onSubmit={[Function]}
      profile={
        Object {
          "dateOfBirth": "10/20/93",
          "emailAddress": "test@gmail.com",
          "emergencyContactName": "Parm Dhillon",
          "emergencyContactNumber": "(408) 315-0213",
          "firstName": "Douglas",
          "gender": "male",
          "lastName": "Dunn",
        }
      }
    />
  </div>
</div>
`;
